File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int mx1=0,mx2=0;
        for(auto num:nums)
        {
            if(num>mx1)
            {
                mx2=mx1;
                mx1=num;
            }
            else
            {
                mx2=max(mx2,num);
            }
        }
        return (mx1-1)*(mx2-1);
    }
};

File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.java
class Solution {
    public int maxProduct(int[] nums) {
        int mx1=0,mx2=0;
        for(int i=0;i<nums.length;i++)
        {
            if(nums[i]>mx1)
            {
                mx2=mx1;
                mx1=nums[i];
            }
            else
            {
                mx2=Math.max(mx2,nums[i]);
            }
        }
        return (mx1-1)*(mx2-1);
    }
}